signal/cosmos.go

```diff
@@ -52,6 +54,9 @@ func (c *Cosmos) getHeaders() map[string]string {
 }
 
 func (c *Cosmos) getEndpoints() []string {
+	if len(c.Endpoints) != 1 {
+		log.Errorf("Cosmos needs 1 endpoint, got %d", len(c.Endpoints))
```

Comment:
this will print out the error but will not fail. Maybe it is better to make `getEndpoints` return an error and then handle the error in the caller code?